liquidity_pool/src/liquidity.rs

```diff
@@ -217,39 +215,55 @@ pub trait LiquidityModule:
         );
         let mut borrow_position = self.borrow_position(borrow_token_nonce).get();
 
-        let accumulated_debt =
-            self.get_debt_interest(borrow_token_amount, borrow_position.timestamp)?;
-        let total_owed = borrow_token_amount + &accumulated_debt;
+        let accumulated_debt = self.get_debt_interest(asset_amount, borrow_position.timestamp)?;
+        let total_owed = asset_amount + &accumulated_debt;
 
         require!(
             asset_amount >= &total_owed,
             "Not enough asset tokens deposited"
         );
```

Comment:
Flawed logic. `total_owned = asset_amount + X`, so the condition will always fail.

The `borrow_token_amount` usage before the change was right. The user pays 1 borrow token for each token they deposited as payment at borrow time.

Reply:
This is true. Updated